### PR TITLE
Add UG filter, nav, linkTo

### PR DESCRIPTION
**Pull request by HollaG**
Fixes 
- #223

### CODE REVIEW

**Comment by HollaG** [comment]
Formatting is not 100%, but I'd like to see how this looks on Markbind

File: docs/UserGuide.md [hunk 1]
```diff
@@ -159,6 +159,108 @@ Failure Output:
...
+![Success for filter 2](images/filter/filter_positive_2.png)
+
+Failure output:
+- When there are invalid values specified in the command `filter -on 2/2/2` (invalid date format)
+![Error for filter](images/filter/filter_negative_1.png)
+```
+Invalid lesson format: 2/2/2 is not a valid date, please use yyyy/mm/dd or mm/dd or dd
+for example, assume today is 2023/11/3, to add 2023/11/29, could use 29, 11/29, 2023/11/29 or 23/11/29. 
+Usage: filter -(any number of unique -[name|subject|before|on|after|remark] [value]). 
+```
```
Comment (pzl111) [inline docs/UserGuide.md:213]: i think jun hiang mentioned that ideally we group these into each student/schedule state

**Review by pzl111 — COMMENTED** [review]
minor comment

File: docs/UserGuide.md (hunk 1 above)
Comment (HollaG) [reply docs/UserGuide.md:213]: Will be resolved when we work on the new unified UG!

**Review by richiehx — APPROVED** [review]
LGTM!

**Pull request merged**